ikarus/solver/nonlinearsolver/newtonraphsonwithscalarsubsidiaryfunction.hh

```diff
@@ -61,7 +54,10 @@ namespace Ikarus {
             })
         : nonLinearOperator_{p_nonLinearOperator},
           linearSolver{std::move(p_linearSolver)},
-          updateFunction{p_updateFunction} {}
+          updateFunction{p_updateFunction} {
+      static_assert(std::tuple_size<typename NonLinearOperatorImpl::ParameterValues>::value == 2,
+                    "The nonlinear operator you pass should only contain the residual and its Jacobian.");
```

Comment:
Can you compare this to `Ikarus::Concepts::LinearSolverCheck` maybe we can write a nice concept for this assertion